Cython/Compiler/Dataclass.py

```diff
@@ -469,6 +470,27 @@ def generate_init_code(code, init, node, fields, kw_only):
     function_start_point.add_code_line(u"def __init__(%s):" % args)
 
 
+def generate_match_args(code, match_args, node, fields, global_kw_only):
+    """
+    Generates a tuple containing what would be the positional args to __init__
+
+    Note that this is generated even if the user overrides init
+    """
+    if not match_args or node.scope.lookup_here("__match_args__"):
+        return
+    positional_arg_names = []
+    for field_name, field in fields.items():
+        # TODO hasattr and global_kw_only can be removed once full kw_only support is added
+        field_is_kw_only = global_kw_only or (
+            hasattr(field, 'kw_only') and field.kw_only.value
```

Comment:
This `hasattr` is only in to anticipate adding a kw_only attribute to field, so that it behaves correctly when that happens. Obviously when that does happen we won't need it.

It's a bit clunky, but it's just future-proofing, and can be removed later